src/zc/lockfile/__init__.py

```diff
@@ -102,3 +103,61 @@ def close(self):
             _unlock_file(self._fp)
             self._fp.close()
             self._fp = None
+
+
+class LockedContext(object):
+    """ContextManager version of LockFile"""
+
+    def __init__(self, name, content_template='{pid}', timeout=float("inf")):
+        self.name = str(name)  # wrap in str() to allow pathlib usage
+        self.content_template = content_template
+        self.lock = None
+        self.timeout = timeout
+
+    def __enter__(self):
+        lockname = self.name + ".lock"
+        tick = 0.1
+        iterations = self.timeout // tick + 1
+        while iterations:
+            try:
+                self.lock = LockFile(lockname, self.content_template)
+                return self._post_lock_operation()
+            except LockError:
+                logger.debug("Couldn't get lock - waiting...")
```

Comment:
It would probably be very helpful if this logging and the exception that are raised just below feature the (absolute) file name being locked. (Probably don't make it absolute when passing it to `LockFile`, just for display to the user?)